src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -98,17 +98,19 @@ private class CriticalHelper
             private ArgBuilder _memberNamesArg;
             private ArgBuilder _memberNamespacesArg;
             private ArgBuilder _collectionContractArg;
-
-            private bool _useReflection = true;
+#endif
 
             public XmlFormatClassReaderDelegate GenerateClassReader(ClassDataContract classContract)
             {
-                if (_useReflection)
+                if (DataContractSerializer.Option == SerializationOption.ReflectionAsBackup || DataContractSerializer.Option == SerializationOption.ReflectionOnly)
                 {
```

Comment:
Right, that's why the default is still set to CodeGenOnly. Will take care of ReflectionAsBackup and make it work in NetCore in a subsequent change.